modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -2148,6 +2168,52 @@ NSFVAction::processVariables()
                      ") supplied to the NavierStokesFV action does not exist!");
 }
 
+bool
+NSFVAction::processThermalConductivity()
+{
+  checkBlockwiseConsistency<MooseFunctorName>("thermal_conductivity_blocks",
+                                              {"thermal_conductivity"});
+  bool have_scalar = false;
+  bool have_vector = false;
+
+  for (unsigned int i = 0; i < _thermal_conductivity_name.size(); ++i)
+  {
+    // First, check if the name is just a number (only in case of isotropic conduction
+    std::istringstream ss(_thermal_conductivity_name[i]);
+    Real real_value;
+    if (ss >> real_value && ss.eof())
+      have_scalar = true;
+    // Now we determine what kind of functor we are dealing with
+    else
+    {
+      if (_problem->hasFunctorWithType<ADReal>(_thermal_conductivity_name[i],
+                                               _problem->parameters().get<THREAD_ID>("_tid")))
+        have_scalar = true;
+      else
+      {
+        if (_problem->hasFunctorWithType<ADRealVectorValue>(
+                _thermal_conductivity_name[i], _problem->parameters().get<THREAD_ID>("_tid")))
+          have_vector = true;
+        else
+        {
+          if (_problem->hasFunctorWithType<Real>(_thermal_conductivity_name[i],
+                                                 _problem->parameters().get<THREAD_ID>("_tid")))
+            have_scalar = true;
+        }
+      }
+    }
+  }
```

Comment:
It looks like we check both `ADReal` and `Real`, but only `ADRealVectorValue`. Should we check for `RealVectorValue` as well? Perhaps we can't process that type, but if that is the case, maybe we can give a helpful error?

Reply:
umm should we complain about Real functor? we ll be missing some derivatives

Reply:
If the functor itself is `Real`, then that implies to me that it doesn't have any derivative information to begin with, so I'm fine not complaining about it. I honestly have not looked into detail at the action. I know that most of our objects take AD functors. I haven't checked to see whether if a user supplies a non-AD functor to the action, we behind-the-scenes convert it to an AD functor for use with the objects.

Reply:
I think I'll disable the Real and RealVectorValue options. I think people can convert functions into PBBFM on the material block if they want to.